linux/LinuxProcessList.c

```diff
@@ -1455,19 +1451,17 @@ static bool LinuxProcessList_recurseProcTree(LinuxProcessList* this, openat_arg_
          LinuxProcessList_readCwd(lp, procFd);
       }
 
-      if (proc->state == 'Z' && (proc->basenameOffset == 0)) {
-         proc->basenameOffset = -1;
-         free_and_xStrdup(&proc->comm, command);
-         lp->procCmdlineBasenameOffset = 0;
-         lp->procCmdlineBasenameEnd = 0;
-         lp->mergedCommand.commChanged = true;
+      if (proc->state == 'Z' && (proc->cmdlineBasenameEnd == 0)) {
+         proc->cmdlineBasenameEnd = -1;
+         free_and_xStrdup(&proc->cmdline, command);
+         proc->cmdlineBasenameStart = 0;
+         proc->mergedCommand.commChanged = true;
```

Comment:
why set `proc->mergedCommand.commChanged` (instead of `cmdlineChanged`), when we have changed `proc->cmdline`?

Reply:
It's the same in the dev branch:
https://github.com/BenBE/htop/blob/master/linux/LinuxProcessList.c#L1458-L1471

`proc->comm` is `proc->cmdline` and still `proc->mergedCommand->commChanged` refers to the process command name, not the command line.